java/src/org/openqa/selenium/manager/SeleniumManagerOutput.java

```diff
@@ -139,6 +143,29 @@ public String getBrowserPath() {
       return browserPath;
     }
 
+    public void validateDriver() {
+      Map<String, String> entries = Map.of("driver", driverPath);
+      validate(entries);
+    }
+
+    public void validateAll() {
+      Map<String, String> entries = Map.of("driver", driverPath, "browser", browserPath);
+      validate(entries);
+    }
+
+    private void validate(Map<String, String> entries) {
+      for (Map.Entry<String, String> entry : entries.entrySet()) {
+        Path filePath = Paths.get(entry.getValue());
+        if (!Files.exists(filePath)) {
+          throw new IllegalArgumentException(
+              "The path for '" + entry.getKey() + "' is not a valid file: " + entry.getValue());
+        } else if (!Files.isExecutable(filePath)) {
+          throw new IllegalArgumentException(
+              "The file at '" + entry.getKey() + "' is not executable: " + entry.getValue());
+        }
+      }
+    }
```

Comment:
Why do we need this logic here? If we need it, shouldn't it be inside Selenium Manager?

Reply:
I'm pulling in checks for what user is passing in as well so they get the same logging and error messaging.